pkg/registry/core/endpoint/storage/storage.go

```diff
@@ -59,3 +68,39 @@ var _ rest.ShortNamesProvider = &REST{}
 func (r *REST) ShortNames() []string {
 	return []string{"ep"}
 }
+
+// Get gets endpoints resource by name in the context (whose significance is tenant and namespace).
+// If the target name starts with prefix of "kubernetes-", and the contextual namespace is "default",
+// this target is considered as "k8s alias", a read-only copy of kubernetes endpoints of default namespace
+// in system tenant, and its content returned is based on the system kubernetes resource.
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	tenant, ok := genericapirequest.TenantFrom(ctx)
+	if !ok {
+		tenant = "system"
+	}
+
+	namespace, ok := genericapirequest.NamespaceFrom(ctx)
+	if !ok {
+		namespace = "default"
+	}
+
+	// redirect default/kubernetes-<network> EP query to default/kubernetes of system tenant
+	if namespace == "default" && strings.HasPrefix(name, "kubernetes-") {
```

Comment:
mke a const for "kubernetes-" ?

Also, have a helper function like IsReadonlyEP(namespace,  name string) bool ?

Reply:
will change as suggested.